state/src/item/metadata.rs

```diff
@@ -124,25 +134,33 @@ impl CacheableItem for Metadata {
 
 const PREFIX: u8 = super::METADATA_PREFIX;
 
+const INITIAL_LEN: usize = 4;
+const TERM_LEN: usize = 2;
+const PARAMS_LEN: usize = 2;
+const TERM_PARAMS_LEN: usize = 1;
+
 impl Encodable for Metadata {
     fn rlp_append(&self, s: &mut RlpStream) {
-        const INITIAL_LEN: usize = 4;
-        const TERM_LEN: usize = 2;
-        const PARAMS_LEN: usize = 2;
-        let mut len = INITIAL_LEN;
-
         let term_changed = self.term != Default::default();
+        let params_changed = self.seq != 0;
+        let term_params_changed = self.term_params.is_some();
```

Comment:
I think this is not an appropriate name for this variable.
It doesn't check if the term param is _changed_, but check if it _exists_.
I think something like `term_params_enabled` would be more appropriate.
Or maybe you can switch the behavior depending on the `era`.